plugin/PDFDocumentPage.qml

```diff
@@ -189,9 +375,182 @@ DocumentPage {
         }
     }
 
+    ContextMenu {
+        id: contextMenuText
+        property variant annotation
+        property point at
+
+        MenuItem {
+            visible: contextMenuText.annotation === undefined
+                     || contextMenuText.annotation === null
+            //% "Add note"
+            text: qsTrId("sailfish-office-me-pdf-txt-anno-add")
+            onClicked: {
+                var annotation = textComponent.createObject(contextMenuText)
+                annotation.color = "#202020"
+                pdfDocument.create(annotation,
+                                   function() {
+                                       var at = view.getPositionAt(contextMenuText.at)
+                                       annotation.attachAt(pdfDocument, at[0], at[2], at[1])
+                                   })
+            }
+        }
+        MenuItem {
+            visible: contextMenuText.annotation !== undefined
+                     && contextMenuText.annotation !== null
+            //% "Edit"
+            text: qsTrId("sailfish-office-me-pdf-txt-anno-edit")
+            onClicked: pdfDocument.edit(contextMenuText.annotation)
+        }
+        MenuItem {
+            visible: contextMenuText.annotation !== undefined
+                     && contextMenuText.annotation !== null
+            //% "Delete"
+            text: qsTrId("sailfish-office-me-pdf-txt-anno-clear")
+            onClicked: contextMenuText.annotation.remove()
+        }
+        Component {
+            id: textComponent
+            PDF.TextAnnotation { }
+        }
+    }
+
+    ContextMenu {
```

Comment:
Can do later, but having this created dynamically when needed would save a bit memory and cpu time on the common case when annotations are not used. Additionally in separate file would save even more as less qml parsing would be needed then.